### PR TITLE
Add longhorn support

### PR DESCRIPTION
**Pull request by ibuildthecloud**

### CODE REVIEW

File: packaging/convoy-agent/launch [hunk 1]
```diff
@@ -37,9 +37,8 @@ storagepool_agent() {
 }
 
 common_vars() {
-    STACK_NAME=$(curl -s http://rancher-metadata/latest/self/stack/name)
-    STACK_UUID=$(curl -s http://rancher-metadata/latest/self/stack/uuid)
-    HOST_UUID=$(curl -s http://rancher-metadata/latest/self/host/uuid)
```
Comment (ibuildthecloud) [inline packaging/convoy-agent/launch:42]: Seemed like HOST_UUID wasn't used anymore, correct?

Comment (cjellick) [inline packaging/convoy-agent/launch:42]: Correct, I missed this reference when i got rid of it everywhere else.

File: packaging/convoy-agent/launch [hunk 2]
```diff
@@ -58,6 +57,7 @@ volume_agent_glusterfs() {
 }
 
 volume_agent_glusterfs_internal() {
+    wait_for_metadata
```
Comment (cjellick) [inline packaging/convoy-agent/launch:62]: Why did you add this? Is it needed for the refactoring you did around separating out convoy + the agent? 
The reason I _didnt_ have it was that volume_agent_glusterfs_internal is called by volume_agent_glusterfs and it is the one that waits on metadata.

**Comment by cjellick** [comment]
```
If you have a good reason for the new wait_for_metadata:
   LGTM
else if you remove it:
  LGMT
else
  DNLGTM
```
